apps/prairielearn/python/prairielearn.py

```diff
@@ -1155,7 +1155,7 @@ def string_fraction_to_number(a_sub, allow_fractions=True, allow_complex=True):
                     "Your expression resulted in a division by zero."
                 )
             except Exception as error:
-                data["format_errors"] = f"Invalid format: {str(error)}"
+                data["format_errors"] = f"Invalid format: {error!s}"
```

Comment:
I believe `!s` isn't necessary for exceptions:

```python
>>> print(f"{Exception('testing')}")
testing
>>> print(f"{Exception('testing')!s}")
testing
```

Reply:
My original hesitancy is that I have seen custom Exceptions override the format specifier. But I don't think we have any of those, so this is probably safe.

Reply:
Resolved in 2732d9ff6340af47d4e4b7f973586eabdf5fc24e.